ngmix/admom/admom.py

```diff
@@ -409,11 +409,19 @@ def get_result(ares):
         res['flux_mean'] = flux_sum/res['wsum']
         res['pars'][5] = res['flux_mean']
 
-    if res['flux_flags'] == 0:
-        if res['flux_err'] > 0:
-            res['s2n'] = res['flux'] / res['flux_err']
-        else:
-            res['flux_flags'] |= ngmix.flags.NONPOS_VAR
+    # if res['flux_flags'] == 0:
+    #     if res['flux_err'] > 0:
+    #         res['s2n'] = res['flux'] / res['flux_err']
+    #     else:
+    #         res['flux_flags'] |= ngmix.flags.NONPOS_VAR
+
+    fc = res['sums_cov'][5, 5]
+    if fc > 0:
+        res['s2n'] = res['sums'][5] / np.sqrt(fc)
+        res['flux_err'] = res['flux'] / res['s2n']
+    else:
+        res['flux_flags'] |= ngmix.flags.NONPOS_VAR
+        res['flags'] |= ngmix.flags.NONPOS_VAR
```

Comment:
Be careful here. The old sums[5] entry needs some factors to make its normalization match other things properly. 

See this comment I had

```
# this is a fun set of factors
            # jacobian area is because ngmix works in flux units
            # the wgt_norm and wsum compute the weighted flux and normalize
            # the weight kernel to peak at 1
            fnorm = jac_area * wgt_norm * res["wsum"]
            res['flux'] = res['sums'][5] / fnorm
```

I think what you have works but it might be worth leaving some comments for future folks.

Reply:
thanks, this is temporary